src/app/extensions.cpp

```diff
@@ -219,8 +219,9 @@ Extension::DitheringMatrixInfo::DitheringMatrixInfo(const std::string& path,
 const render::DitheringMatrix& Extension::DitheringMatrixInfo::matrix() const
 {
   if (!m_loaded) {
+    if (!load_dithering_matrix_from_sprite(m_path, m_matrix))
+      throw std::runtime_error("Invalid matrix name");
```

Comment:
I think we have two options:

1. `load_dithering_matrix_from_sprite()` could throw the exception directly (instead of returning a bool), or 
2. adding a new bool `DitheringMatrixInfo::loadMatrix()` or `render::DitheringMatrix::isValidMatrix()` to avoid the exception at all (I think I prefer this one)

In this way we can create each `DitherItem` only for valid matrices (ignoring invalid matrices). With this probably we'd need in `render/dithering_matrix.h` something like:

```c++
  class DitheringMatrix {
  public:
    ...
    static DitheringMatrix InvalidMatrix() { return DitheringMatrix(0, 0); }

    bool isValid() const { return m_rows > 0 && m_cols > 0; }
    ...
```

Reply:
From the `load_dithering_matrix_from_sprite` impl, it looks like the only way that it returns `false` is because the filename doesn't exist, if we keep the exception at this point, I'd change the error message to something like `fmt::format("The dithering matrix file {} doesn't exist", m_path)`.

Reply:
I kept the exception and took your recommendation on the error message.